packages/teams-js/src/public/pages.ts

```diff
@@ -1096,4 +1103,117 @@ export namespace pages {
         : false;
     }
   }
+
+  /**
+   * Provides APIs to interact with the button in hosts
+   *
+   * @beta
+   */
+  export namespace responseButton {
+    /**
+     * The required information that the app needs to let host apps know.
+     *
+     * @beta
+     */
+    export interface ResponseInfo {
+      /**
+       * The unique ID helps the host and apps identify which button it refers to.
+       *
+       * @beta
+       */
+      responseId: string;
+      /**
+       * Id of File on OneDrive
+       *
+       * @beta
+       */
+      oneDriveFileId: string;
+      /**
+       * The exact message where the action was invoked from
+       *
+       * @beta
+       */
+      messageId: string;
+
+      /**
+       * The conversation where the action was invoked from
+       *
+       * @beta
+       */
+      conversionId: string;
+    }
+
+    /**
+     * Notify host apps to show the response button.
+     * @param params - Parameters sent to host apps contain the required information.
+     *
+     * @beta
+     */
+    export function showResponseButton(params: ResponseInfo): Promise<void> {
+      ensureInitialized(runtime, FrameContexts.content);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (!params) {
+        throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      }
+      return sendAndHandleSdkErrorWithVersion(
+        getApiVersionTag(pagesTelemetryVersionNumber, ApiName.Pages_ResponseButton_ShowResponseButton),
+        'pages.responseButton.showResponseButton',
+        params,
+      );
+    }
+
+    /**
+     * Notify host apps to hide the response button
+     *
+     * @beta
+     */
+    export function hideResponseButton(): Promise<void> {
+      ensureInitialized(runtime, FrameContexts.content);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      return sendAndHandleSdkErrorWithVersion(
+        getApiVersionTag(pagesTelemetryVersionNumber, ApiName.Pages_ResponseButton_HideResponseButton),
+        'pages.responseButton.hideResponseButton',
+      );
+    }
+
+    /**
+     * Registers a handler for when the button in the host apps is clicked by the user
+     * @param appEventHandler - The handler to invoke when the user clicks on the button in host apps.
+     *
+     * @beta
+     */
+    export function responseButtonEventHandler(appEventHandler: handlerFunctionType): void {
```

Comment:
Is there no data the host has to pass down to the app developer when this handler is triggered? For example, if the response button is a button with a dropdown that presents multiple options (forward, replyAll, etc.) do you need to make an enum or something similar available here?

Reply:
I remember from earlier discussion(like a year ago) we did not to want to maintain a list of enums (like: reply, replyall, forward), due to the reason that if hubs decide to add more dropdown options, we will need to add to this list and needs SDK changes. So, it is hard to scale and maintain in the long term. 

If we want to allow developers know which button got clicked, and able to handle different logic based on which button got clicked. That sounds good to me. My only ask is that developer don't need to register a handler for each of the dropdown item if they share the same handler logic.